src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -572,7 +574,19 @@ public function text(/* $default = null */)
             throw new \InvalidArgumentException('The current node list is empty.');
         }
 
-        return $this->getNode(0)->nodeValue;
+        $text = $this->getNode(0)->nodeValue;
+
+        if (\func_num_args() > 1) {
+            if (func_get_arg(1)) {
+                $text = preg_replace("/(\n|\t)/", '', $text);
+
+                return preg_replace('/ {2,}/', ' ', $text);
```

Comment:
don't we trim left and right?

Reply:
I'd suggest removing the above preg_replace and do this here:
`return trim(preg_replace('/\s++/', ' ', $text));`

Reply:
* `preg_replace("/(\n|\t)/", '', $text)` is erroneous: it should replace with a space
* about performances, see on CodeIgniter: [#1750](https://github.com/bcit-ci/CodeIgniter/pull/1750) / [62a5ee3](https://github.com/bcit-ci/CodeIgniter/commit/62a5ee3e430bd18a8ca8afa0d704967e8bd25763)

Reply:
about performance, this looks even faster: `'/(?:\s{2,}+|[^\S ])/'`

Reply:
It's so obscure, but for the sake of maximum performance. I like this.

Especially the double negation for "space characters minus the regular space".

Reply:
very nice